debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerStreamingChangeEventSource.java

```diff
@@ -83,98 +99,195 @@ public void execute(ChangeEventSourceContext context) throws InterruptedExceptio
+        if (m.matches()) {
+            final String captureName = m.group(1);
+            LOGGER.info("Table is no longer captured with capture instance {}", captureName);
+            return Arrays.asList(currentChangeTables).stream()
+                    .filter(x -> !x.getCaptureInstance().equals(captureName))
+                    .collect(Collectors.toList()).toArray(new ChangeTable[0]);
+        }
+        throw exception;
+    }
+
+    private ChangeTable[] getCdcTablesToQuery() throws SQLException, InterruptedException {
+        final Set<ChangeTable> cdcEnabledTables = connection.listOfChangeTables();
+
+        final Map<TableId, List<ChangeTable>> whitelistedCdcEnabledTables = cdcEnabledTables.stream()
+                .filter(changeTable -> {
+                    if (connectorConfig.getTableFilters().dataCollectionFilter().isIncluded(changeTable.getSourceTableId())) {
+                        return true;
+                    }
+                    else {
+                        LOGGER.info("CDC is enabled for table {} but the table is not whitelisted by connector");
+                        return false;
+                    }
+                })
+                .collect(Collectors.groupingBy(x -> x.getSourceTableId()));
+
+        final List<ChangeTable> tables = new ArrayList<>();
+        for (List<ChangeTable> captures: whitelistedCdcEnabledTables.values()) {
+            ChangeTable changeTable = captures.get(0);
+            if (captures.size() > 1) {
+                if (captures.get(0).getStartLsn().compareTo(captures.get(1).getStartLsn()) < 0) {
+                    captures.get(0).setStopLsn(captures.get(1).getStartLsn());
+                    tables.add(captures.get(1));
+                }
+                else {
+                    captures.get(1).setStopLsn(captures.get(0).getStartLsn());
+                    changeTable = captures.get(1);
+                    tables.add(captures.get(0));
+                }
+                LOGGER.info("Multiple capture instances {} and {} present for the same table", changeTable, captures.get(1));
+            }
+            if (schema.tableFor(changeTable.getSourceTableId()) == null) {
+                LOGGER.info("Table {} is new to be monitored by capture instance {}", changeTable.getSourceTableId(), changeTable.getCaptureInstance());
+                // We need to read the source table schema - primary key information cannot be obtained from change table
+                dispatcher.dispatchSchemaChangeEvent(changeTable.getSourceTableId(), new SqlServerSchemaChangeEventEmitter(offsetContext, changeTable, connection.getTableSchemaFromTable(changeTable)));
+            }
+            tables.add(changeTable);
+        }
+
+        return tables.toArray(new ChangeTable[tables.size()]);
+    }
+
     @Override
     public void commitOffset(Map<String, ?> offset) {
     }
 
-    private static class ChangeTable {
+    private static class ChangeTablePointer {
```

Comment:
What's the purpose of this class? Please add a comment. Also perhaps `ChangeTableEntry`?